### PR TITLE
Fix/signup api for frontend

### PR DESCRIPTION
**Pull request by azsidhu**

### CODE REVIEW

File: digital_benchmark/accounts/views.py [hunk 1]
```diff
@@ -51,9 +53,39 @@ def post(self, request):
...
         logout(request)
-        return redirect('/')
\ No newline at end of file
+        return redirect('/')
+
+
+class Signup(APIView):
+    def post(self, request, format=None):
+        try:
+            user = User.objects.get(username=request.data.get('username', ''))
```
Comment (UsmanAbbasi1) [inline digital_benchmark/accounts/views.py]: Validate api input params through serializer. Also add UniqueValidator for 'username' field.

File: digital_benchmark/accounts/views.py [hunk 2]
```diff
@@ -51,9 +53,39 @@ def post(self, request):
...
+                }
+            except User.DoesNotExist:
+                user, created = User.objects.get_or_create(email=request.data.get(
+                    'email', ''), username=request.data.get('username', ''))
+                if not created:
+                    response = {
+                        'error_messages': ['User not created.', ]
+                    }
+                else:
+                    user.set_password(request.data.get('password', ''))
```
Comment (UsmanAbbasi1) [inline digital_benchmark/accounts/views.py]: see why we use 'set_password' to set password.

File: digital_benchmark/accounts/views.py [hunk 3]
```diff
@@ -51,9 +55,14 @@ def post(self, request):
...
 class LogoutView(View):
     def get(self, request):
         logout(request)
-        return redirect('/')
\ No newline at end of file
+        return redirect('/')
+
+
+class RegisterUserView(generics.CreateAPIView):
+    serializer_class = UserSerializer
```
Comment (UsmanAbbasi1) [inline digital_benchmark/accounts/views.py]: @azsidhu see what happens with password field when user is created through this endpoint. Does password gets saved as hashed value or plain value.

**Comment by UsmanAbbasi1** [comment]
@azsidhu You have conflicts in this PR.
Add '.gitignore' file in your project.

File: digital_benchmark/accounts/views.py (hunk 3 above)
Comment (azsidhu) [reply digital_benchmark/accounts/views.py]: It's being saved as hashed value, because I'm overriding create method in UserSerializer.

**Pull request merged**